### ISSUE TITLE
API for uploading avatars

### ISSUE THREAD

**Issue by pdpinch** [mitodl/micromasters#394]
update the model to store an image 

Use django-storages back-end to manage location for saving

Same authorizations as profile API

**Comment by pdpinch** [comment]
@gsidebo before you work on this, I'd like to talk about the possibility of using the gravatar service.

**Comment by gsidebo** [comment]
@pdpinch sounds good. im going to take a swing at this after #369 and #375

**Comment by giocalitri** [comment]
just for my own knowledge I try to implement the support for the gravatar images ([docs here](https://en.gravatar.com/site/implement/images/)) and I came out with this code for the profile serializer:

```
class ProfileSerializer(ModelSerializer):
    """Serializer for Profile objects"""
    pretty_printed_student_id = SerializerMethodField()
    profile_url_full = SerializerMethodField()
    work_history = EmploymentSerializer(many=True)
    education = EducationSerializer(many=True)

    def get_pretty_printed_student_id(self, obj):
        """helper method for the SerializerMethodField"""
        # pylint: disable=no-self-use
        return obj.pretty_printed_student_id

    def get_profile_url_full(self, obj):
        """"""
        if obj.has_profile_image:
            return obj.profile_url_full
        # otherwise return the gravatar version
        md5_hash = hashlib.md5()
        md5_hash.update(obj.user.email.lower().encode())
        return "http://www.gravatar.com/avatar/{0}?d={1}".format(
            md5_hash.hexdigest(),
            parse.quote_plus(obj.profile_url_full)
        )
```

The only problem with this is that the `obj.profile_url_full` must be a public URL because it will be served by gravatar and not by edx directly, so it does not work in development environment.
A solution might be to always point to the edX default image public URL.

Anyway: TL;DR: it is pretty easy to implement.

**Comment by giocalitri** [comment]
We are going to postpone this story.

**Comment by pdpinch** [comment]
Rolled into #179
